### PR TITLE
Make the is_nested() function check more toroidal planes for non-axis…

### PR DESCRIPTION
**Pull request by dpanici**
…ymmetric equilibria, as it was possible for the zeta=0 plane to be nested but the other zeta planes to become un-nested

### CODE REVIEW

File: desc/configuration.py [hunk 1]
```diff
@@ -1449,7 +1449,14 @@ def body_fun(k_rhok_thetak_Rk_Zk):
 
         return jnp.vstack([rho, theta, phi]).T
 
-    def is_nested(self, nsurfs=10, ntheta=20, zeta=0, Nt=45, Nr=20):
+    def is_nested(
+        self,
+        nsurfs=10,
+        ntheta=20,
+        zetas=np.linspace(0, 2 * np.pi, 4, endpoint=False),
```
Comment (f0uriest) [inline desc/configuration.py:1456]: might make more sense to make this a scalar `nzeta` to be consistent with the others. What I've done for others is have it default to `None` and that gets treated as "zeta=0 for N=0, else zeta=range..."

File: desc/configuration.py [hunk 2]
```diff
@@ -1471,33 +1478,38 @@ def is_nested(self, nsurfs=10, ntheta=20, zeta=0, Nt=45, Nr=20):
...
-        Zv = v_coords["Z"].reshape((t_grid.L, t_grid.M, t_grid.N))[:, :, 0]
-
-        rline = MultiLineString(
-            [LineString(np.array([R, Z]).T) for R, Z in zip(Rr, Zr)]
-        )
-        vline = MultiLineString(
-            [LineString(np.array([R, Z]).T) for R, Z in zip(Rv.T, Zv.T)]
-        )
+        planes_nested_bools = []
+        if self._N == 0:
```
Comment (f0uriest) [inline desc/configuration.py:1482]: see previous comment, its probably best not to override what the user gives as input (if for some reason they want to check 10 toroidal planes for an axisymmetric case, that's fine).

Also in general try to use `self.N` rather than `self._N` when reading values, that way any logic about what value to return will work correctly

**Review by f0uriest — APPROVED** [review]
